submissions/pozzitive11/js-world-oop/index.js

```diff
@@ -0,0 +1,78 @@
+class Inhabitant {
+  constructor(species, name, gender, saying) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.saying = saying;
+  }
+
+  getProperties() {
+    return `${this.species}; My name ${this.name}; ${this.gender}; I have ${this.legs} legs and ${this.hands} hands; ${this.saying};` ;
+  }
+}
+
+class Animals extends Inhabitant {
+  constructor(species, name, gender, paw, saying) {
+    super(species, name, gender, saying);
+    this.paw = paw;
+  }
+  getProperties() {
+    return `${this.species}; ${this.name}; ${this.gender}; I have ${this.paw} paws; ${this.saying};`;
+  }
+}
+
+class Humans extends Inhabitant {
+  constructor(species, name, gender, legs, hands, saying){
+    super(species, name, gender, saying);
+    this.legs = legs;
+    this.hands = hands;
+  }
+}
+
+class Dog extends Animals {
+  constructor(species, name, gender, saying, paw) {
+    super(species, name, gender, paw, saying);
+  }
+}
+
+class Cat extends Animals {
+  constructor(species, name, gender, paw) {
+    super(species, name, gender, paw, "mew");
+  }
+}
+
+class Catwoman extends Cat {
+  constructor(species, name, gender, paws, saying) {
+    super(species, name, gender, paws, saying);
+  } 
+}
+
+class Man extends Humans {
+  constructor(species, name, gender, legs, hands, saying) {
+    super(species, name, gender, legs, hands, saying);
```

Comment:
We want all men (and women) be humans. Why would we grant anyone the legal opportunity to override the species here? You may pass scalar values to a super's constructor, not just simply hand over some params.